server.go

```diff
@@ -1227,6 +1229,43 @@ func newServer(cfg *Config, listenAddrs []net.Addr, chanDB *channeldb.DB,
 		}
 	}
 
+	if len(cfg.ExternalHosts) != 0 {
+		s.hostAnn = netann.NewHostAnnouncer(netann.HostAnnouncerConfig{
+			Hosts:         cfg.ExternalHosts,
+			RefreshTicker: ticker.New(time.Minute * 5),
+			LookupHost: func(host string) (net.Addr, error) {
+				return lncfg.ParseAddressString(
+					host, strconv.Itoa(defaultPeerPort),
+					cfg.net.ResolveTCPAddr,
+				)
+			},
+			AnnounceNewIPs: func(newAddrs []net.Addr, oldAddrs map[string]net.Addr) error {
+				_, err := s.genNodeAnnouncement(
+					true, func(currentNodeAnn *lnwire.NodeAnnouncement) {
```

Comment:
this closure could be extracted to `netann` package for readability. it doesn't depend on anything in the server.

Reply:
Done.